fastchat/model/falcon_model.py

```diff
@@ -0,0 +1,133 @@
+import torch
+from typing import Iterable
+import gc
+from threading import Thread
+from transformers import TextIteratorStreamer, GenerationConfig
+
+import transformers
+transformers.logging.set_verbosity_error()
+
+def partial_stop(output, stop_str):
+    for i in range(0, min(len(output), len(stop_str))):
+        if stop_str.startswith(output[-i:]):
+            return True
+    return False
+
+@torch.inference_mode()
+def falcon_generate_stream(
+    model, tokenizer, params, device, context_len=2048, stream_interval=2
+):
```

Comment:
Can you write some comments on why we need a separate `generate_stream` for falcon? and how is it different from our normal one.

Reply:
I am experiencing a similar issue as described in this GitHub link: https://github.com/lm-sys/FastChat/issues/1504#issuecomment-1567832179. When attempting to perform inference using the default generate stream in FastChat, I am encountering unexpected results. However, when I utilize the .generate function provided by Hugging Face directly, I receive the expected output. As a solution, we have created a new generate stream based on Hugging Face's .generate function and integrated it into the FastChat generate stream implementation.

Reply:
The biggest difference is that the main generation work is finished by  `.generate` function provided by Hugging Face directly